src/protagonist/API/Features/NamedQueries/Requests/CreateNamedQuery.cs

```diff
@@ -0,0 +1,57 @@
+﻿using API.Infrastructure.Requests;
+using DLCS.Core;
+using DLCS.Model.Assets.NamedQueries;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Features.NamedQueries.Requests;
+
+public class CreateNamedQuery : IRequest<ModifyEntityResult<NamedQuery>>
+{
+    public int CustomerId { get; }
+    
+    public NamedQuery NamedQuery { get; }
+    
+    public CreateNamedQuery(int customerId, NamedQuery namedQuery)
+    {
+        CustomerId = customerId;
+        NamedQuery = namedQuery;
+    }
+}
+
+public class CreateNamedQueryHandler : IRequestHandler<CreateNamedQuery, ModifyEntityResult<NamedQuery>>
+{  
+    private readonly DlcsContext dbContext;
+
+    public CreateNamedQueryHandler(DlcsContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+    
+    public async Task<ModifyEntityResult<NamedQuery>> Handle(CreateNamedQuery request, CancellationToken cancellationToken)
+    {
+        var existingNamedQuery = await dbContext.NamedQueries.SingleOrDefaultAsync(
```

Comment:
Can use `.AsNoTracking()`, we don't need to have change tracking for this

Reply:
Fixed in 2c2b4ae23af0d13c34576cd0c8ed7b2f8a6c6594